test/cppsuite/test_harness/workload/thread_context.cxx

```diff
@@ -285,6 +285,45 @@ thread_context::insert(
     return (true);
 }
 
+bool
+thread_context::remove(
+  scoped_cursor &cursor, uint64_t collection_id, const std::string &key, wt_timestamp_t ts)
+{
+    WT_DECL_RET;
+    testutil_assert(tracking != nullptr);
+    testutil_assert(cursor.get() != nullptr);
+
+    /*
+     * When no timestamp is specified, get one to apply to the update. We still do this even if the
+     * timestamp manager is not enabled as it will return a value for the tracking table.
+     */
+    if (ts == 0)
+        ts = tsm->get_next_ts();
+    transaction.set_commit_timestamp(ts);
+
+    cursor->set_key(cursor.get(), key.c_str());
+    ret = cursor->remove(cursor.get());
+    if (ret != 0) {
+        if (ret == WT_ROLLBACK) {
+            transaction.set_needs_rollback(true);
+            return (false);
+        } else
+            testutil_die(ret, "unhandled error while trying to remove a key");
+    }
+    ret = tracking->save_operation(
+      tracking_operation::DELETE_KEY, collection_id, key.c_str(), "", ts, op_track_cursor);
+    if (ret != 0) {
+        if (ret == WT_ROLLBACK) {
+            transaction.set_needs_rollback(true);
+            return (false);
+        } else
+            testutil_die(
+              ret, "unhandled error while trying to save an remove to the tracking table");
```

Comment:
a remove?

Reply:
From my understanding, the tracking table saves all the operations into a WT table that happens in the test. This is so that we can keep a log of each operation in the table for the validation phase. 

I learnt recently that the validation phase happens at the end of each cppsuite test, and uses the tracking table to reconstruct an in-memory representation of how the table is meant to look like, and further reconciles with the actual table in the test. @etienneptl and @luke-pearson is my understanding correct?

Reply:
The validation takes place only if configured which I believe is true by default. Yes, your understanding is correct.
"an remove" -> "a remove/removal/deletion", choose a word but fix the "a".